shardingsphere-features/shardingsphere-authority/shardingsphere-authority-common/src/main/java/org/apache/shardingsphere/authority/loader/storage/impl/dialect/OraclePrivilegeLoader.java

```diff
@@ -0,0 +1,189 @@
+        for (Entry<ShardingSphereUser, List<PrivilegeType>> entry : privilegeCache.entrySet()) {
+            userPrivilegeMap.get(entry.getKey()).getAdministrativePrivileges().getPrivileges().addAll(entry.getValue());
+        }
+    }
+
+    private void collectSysPrivileges(final Map<ShardingSphereUser, List<PrivilegeType>> privilegeCache, final ResultSet resultSet) throws SQLException {
+        String privilegeType = resultSet.getString("PRIVILEGE");
+        String grantee = resultSet.getString("GRANTEE");
+        privilegeCache
+                .computeIfAbsent(new ShardingSphereUser(grantee, "", ""), k -> new ArrayList<>())
+                .add(getPrivilegeType(privilegeType));
+    }
+
+    private Optional<ShardingSphereUser> findShardingSphereUser(final Map<ShardingSphereUser, ShardingSpherePrivileges> userPrivilegeMap, final ResultSet resultSet) throws SQLException {
+        Grantee grantee = new Grantee(resultSet.getString("rolname"), "");
+        return userPrivilegeMap.keySet().stream().filter(each -> each.getGrantee().equals(grantee)).findFirst();
+    }
+
+    private String getSysPrivilegesSQL(final Collection<ShardingSphereUser> users) {
+        String userList = users.stream().map(each -> String.format("'%s'", each.getGrantee().getUsername())).collect(Collectors.joining(", "));
+        return String.format(SYS_PRIVILEGE_SQL, userList);
+    }
+
+    private String getTablePrivilegesSQL(final Collection<ShardingSphereUser> users) {
+        String userList = users.stream().map(each -> String.format("'%s'", each.getGrantee().getUsername())).collect(Collectors.joining(", "));
+        return String.format(TABLE_PRIVILEGE_SQL, userList);
+    }
+
+    private PrivilegeType getPrivilegeType(final String privilege) {
+        switch (privilege) {
+            case "SELECT":
+                return PrivilegeType.SELECT;
+            case "INSERT":
+                return PrivilegeType.INSERT;
+            case "UPDATE":
+                return PrivilegeType.UPDATE;
+            case "DELETE":
+                return PrivilegeType.DELETE;
+            case "REFERENCES":
+                return PrivilegeType.REFERENCES;
+            case "INDEX":
+                return PrivilegeType.INDEX;
+            case "EXECUTE":
+                return PrivilegeType.EXECUTE;
+            case "CREATE PROCEDURE":
+                return PrivilegeType.CREATE_PROC;
+            case "CREATE ROLE":
+                return PrivilegeType.CREATE_ROLE;
+            case "CREATE SEQUENCE":
+                return PrivilegeType.CREATE_SEQUENCE;
+            case "CREATE TABLESPACE":
+                return PrivilegeType.CREATE_TABLESPACE;
+            case "CREATE USER":
+                return PrivilegeType.CREATE_USER;
+            case "CREATE VIEW":
+                return PrivilegeType.CREATE_VIEW;
+            case "SYSDBA":
+                return PrivilegeType.SUPER;
+            // TODO other privilege
```

Comment:
So these todo privilege is expected in next PR?

Reply:
yes，the privilege type of oracle is too many.